### ISSUE TITLE
About instance normalization in LinearAttention

### ISSUE THREAD

**Issue by junfeng-chen** [BaratiLab/OFormer#3]
Hi Zijie,

I managed to run your code on my machine. Very nice work and code ! 

I am using `torch==2.3`. When running the Darcy example, I encounted an issue with the `nn.InstanceNorm1d` founded in `LinearAttention` and `CrossLinearAttention`. The training goes on, but with a warning on dimension mismatch.

In the `norm_wrt_domain` function, you reshape the tensor to `(bh) n c` before applying the instance normalization function `nn.InstanceNorm1d(dim_head)`. However, Pytorch suggests the input shape to be `(bh) c n`. 

Can you have a look ? https://pytorch.org/docs/stable/generated/torch.nn.InstanceNorm1d.html

**Comment by zijieli-Jlee** [comment]
Hi Junfeng, 
Thanks for your interest in this work.  Actually you were right about this, when carrying out the experiment in the paper I used  this masked_instance_norm https://github.com/BaratiLab/OFormer/blob/f1111b3c634893329e14e47e8903e421dd726a0c/BVP/nn_module/attention_module.py#L79 which assumes input last dimension is channel dimension, but I forgot to rearrange the layout after changing to ```nn.InstanceNorm```, for a more up-to-date implementation please check this one: https://github.com/zijieli-Jlee/neuraloperator-dev/blob/dev/neuralop/layers/attention_kernel_integral.py.

**Comment by junfeng-chen** [comment]
> Hi Junfeng, Thanks for your interest in this work. Actually you were right about this, when carrying out the experiment in the paper I used this masked_instance_norm
> 
> https://github.com/BaratiLab/OFormer/blob/f1111b3c634893329e14e47e8903e421dd726a0c/BVP/nn_module/attention_module.py#L79
> 
> which assumes input last dimension is channel dimension, but I forgot to rearrange the layout after changing to `nn.InstanceNorm`, for a more up-to-date implementation please check this one: https://github.com/zijieli-Jlee/neuraloperator-dev/blob/dev/neuralop/layers/attention_kernel_integral.py.

Hi Zijie,

Thanks for the clarification and for pointing me to the updated implementation. This resolves my issue.
